### PR TITLE
Move CI to GitHub Actions

### LINKED ISSUE #9: Move CI to GitHub Actions
**What is missing?**"

Instead of GitLab CI for the project, GitHub Actions should be used instead. 

**Why do we need it?**

Users outside of AZ cannot see the results of their pipelines. 

Additionally, we can implement requirements for status checks to pass and add additional security measures.

### PR DESCRIPTION
**Pull request by seancallaway**
*Issue #:* #9

*Description of changes:*

- Remove unused Travis CI file.
- Adjust build commands in Dockerfile.
- Add test for Go.
- Add linting for Helm charts.
- Add container build/push for metric adapter and consumer.
- Add build/push for Helm charts.
- Remove GitLab CI file and integration.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: .github/workflows/build_consumer.yaml [hunk 1]
```diff
@@ -0,0 +1,40 @@
+name: Build and Push Adapter Container
```
[the remark this answers is not in the document]
Comment (seancallaway) [reply .github/workflows/build_consumer.yaml:1]: Yes! Good catch.

File: Dockerfile [hunk 2]
```diff
@@ -2,7 +2,8 @@ FROM golang:1.14-alpine as builder
 RUN apk add -U --no-cache ca-certificates
 WORKDIR ${GOPATH}/src/github.com/awslabs/k8s-cloudwatch-adapter
 COPY . ./
-RUN CGO_ENABLED=0 GOOS=linux go build -mod=vendor -tags=netgo -o /adapter cmd/adapter/adapter.go
+RUN CGO_ENABLED=0 GOOS=linux go mod vendor \
```
[the remark this answers is not in the document]
Comment (seancallaway) [reply Dockerfile:5]: I swear I fixed that. Maybe I didn't push? Either way, will resolve that.

[the remark this answers is not in the document]
Comment (seancallaway) [reply Dockerfile]: Fixed via a push.

File: .github/workflows/build_consumer.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (seancallaway) [reply .github/workflows/build_consumer.yaml]: Fixed!

**Review by kody-actzero — APPROVED** [review]

**Pull request merged**